lib/msal-common/src/account/AccountInfo.ts

```diff
@@ -83,10 +81,16 @@ export function buildTenantProfileFromIdTokenClaims(
 ): TenantProfile {
     const { oid, sub, tid, name, tfp, acr } = idTokenClaims;
 
+    /**
+     * Since there is no way to determine if the authority is AAD or B2C, we exhaust all the possible claims that can serve as tenant ID with the following precedence:
+     * tid - TenantID claim that identifies the tenant that issued the token in AAD. Expected in all AAD ID tokens, not present in B2C ID Tokens.
+     * tfp - Trust Framework Policy claim that identifies the policy that was used to authenticate the user. Functions as tenant for B2C scenarios.
+     * acr - Authentication Context Class Reference claim used only with older B2C policies. Fallback in case tfp is not present, but likely won't be present anyway.
```

Comment:
I think acr is not B2C specific. Can be returned by other authorities but not sure under what conditions